src/identity_policy_provider.py

```diff
@@ -0,0 +1,208 @@
+            "properties": {
+                "Statement": {
+                    "type": "array",
+                    "items": {
+                        "type": "object",
+                        "required": ["Effect", "Principal", "Action"],
+                        "properties": {
+                            "Effect": {
+                                "type": "string",
+                                "enum": ["Allow", "Deny"]
+                            },
+                            "Principal": {
+                                "type": "object",
+                                "oneOf": [{
+                                    "required": ["AWS"],
+                                    "properties": {
+                                        "AWS": {
+                                            "type": "string"
+                                        }
+                                    }
+                                }, {
+                                    "required": ["AWS"],
+                                    "properties": {
+                                        "AWS": {
+                                            "type": "array",
+                                            "items": {
+                                                "type": "string"
+                                            }
+                                        }
+                                    }
+                                }]
+                            },
+                            "Action": {
+                                "type": "array",
+                                "items": {
+                                    "type": "string",
+                                    "enum": ["ses:SendEmail", "ses:SendRawEmail"]
+                                }
+                            }
+                        }
+                    }
+                },
+                "Version": {
+                    "type": "string",
+                    "description": "of the policy document"
+                }
+            }
+        }
+    }
+}
+
+
+class IdentityPolicyProvider(ResourceProvider):
+
+    def __init__(self):
+        super(IdentityPolicyProvider, self).__init__()
+        self.request_schema = request_schema
+        self.ses = boto3.client('ses')
+
+    def convert_property_types(self):
+        self.heuristic_convert_property_types(self.properties)
+
+    def create(self):
+        existing_policy = self.get_policy(self.get('Identity'), self.get('PolicyName'))
+        if existing_policy is not None:
+            self.fail(f"identity policy {self.get('PolicyName')} already exists")
+            return
+
+        desired_policy_document = PolicyDocument.from_dict(self.get('PolicyDocument'))
+        self.put_policy(desired_policy_document)
+
+    def update(self):
+        if self.physical_resource_id != self.get('PolicyName'):
+            self.fail("changing property PolicyName is not allowed")
```

Comment:
If you cannot update the PolicyName, then you also should prevent updating the Identity.

```python
if self.policy_name != self.old_policy_name or self.identity != self.old_identity:
   self.fail(...)
```

Alternatively, I recommend that changing the identity and/or policy name is treated as the insert of a new one.
By returning a different physical_resource_id CFN will respond with a delete request for the old policy.

```python
if self.policy_name != self.old_policy_name or self.identity != self.old_identity:
   self.insert()
   return
```

Reply:
Of course, I should have known that, thanks!